resource-aggregate/events/resourceChanged.go

```diff
@@ -90,6 +96,9 @@ func (rc *ResourceChanged) Equal(changed *ResourceChanged) bool {
 	if rc.GetAuditContext().GetUserId() != changed.GetAuditContext().GetUserId() {
 		return false
 	}
+	if len(changed.GetResourceTypes()) > 0 && !slices.Equal(rc.GetResourceTypes(), changed.GetResourceTypes()) {
```

Comment:
What if `len(changed.GetResourceTypes()) == 0` and `len(rc.GetResourceTypes()) > 0`, that cannot occur or why is it this case ignored as change by this condition?

Reply:
fixed